### ISSUE TITLE
Allow configuration to remove sensitive data from flags responses

### ISSUE THREAD

**Issue by matthewelwell** [SolidStateGroup/bullet-train-api#1255]
We should add a configuration (at the environment or project level?) to remove descriptions from the response. Alternatively, we could add another field to the feature, in which you can store more sensitive information.

**Comment by gagantrivedi** [comment]
Is this for SDK endpoints? If yes - I don't think we need it since we have removed this field entirely in edge, no?

**Comment by dabeeeenster** [comment]
Yes but for Core and Self Hosted people this would be helpful

**Comment by matthewelwell** [comment]
At the moment, the responses to flags (from the Core API) includes the feature `description` and `tags` fields. These are not necessary to return and potentially sensitive.

**Comment by gagantrivedi** [comment]
Implemented here: https://github.com/Flagsmith/flagsmith/pull/2017
NOTE: please read the discussion on the above PR for more context

Tldr: a flag named `hide_sensitive_data` has been added to the environment endpoint that(if enabled) will make the `flags` response looks like the response from edge-api and will also not return traits for `identity` endpoints

**Comment by matthewelwell** [comment]
@gagantrivedi I think this is just waiting the frontend piece, is that correct?

**Comment by novakzaballa** [comment]
@matthewelwell, For the frontend changes, should we request the endpoint to hide sensitive data solely based on the value of the `hide_sensitive_data` feature flag, or should we allow the admins to set this behavior somewhere as you proposed in the first comment?

**Comment by gagantrivedi** [comment]
> @matthewelwell, For the frontend changes, should we request the endpoint to hide sensitive data solely based on the value of the `hide_sensitive_data` feature flag, or should we allow the admins to set this behavior somewhere as you proposed in the first comment?

Yeah, we went with the first approach, and it is now a configuration at the environment level

**Comment by matthewelwell** [comment]
@gagantrivedi I think this still needs implementing in the Edge API? For example, the traits need removing on the identities endpoints. Also, we should add a PR for the docs to confirm which fields are removed on which endpoints.

**Comment by gagantrivedi** [comment]
Yeah, it's here: https://github.com/Flagsmith/edge-api/pull/145

**Comment by kyle-ssg** [comment]
@novakzaballa I think this can be closed?

**Comment by novakzaballa** [comment]
Yes, we can close it

**Comment by BioSurienDG** [comment]
I tried to toggle this in our self-hosted app but  after I click confirm the toggle isn't updated?
